### PR TITLE
refactor(envelope): more tests/docs and functions not a type

### PR DESCRIPTION
**Pull request by smoyer64**
This PR provides an implementation of the UCAN token's wire-protocol `Envelope` as described in https://github.com/ucan-wg/spec#envelope.

This code is fully functional but further non-breaking work might include:

- Actually generating the CID (the functions are currently returning `cid.Undef` to establish a fixed exported API.
- Eliminating requirement #2 for types that wish to be enclosed in an Envelope.
- Turning at least some of the `error`s into `var`s and wrapping some that return obscure wording.

### CODE REVIEW

File: internal/envelope/ipld.go [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+		if err != nil {
+			return undef, cid.Undef, err
+		}
+
+		kStr, err := k.AsString()
+		if err != nil {
+			return undef, cid.Undef, fmt.Errorf("the SigPayload keys are not strings: %w", err)
+		}
+
+		keyCount++
```
Comment (MichaelMure) [inline internal/envelope/ipld.go:167]: you could stop trying early if there is too much keys

File: internal/envelope/example_test.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+func (e *Example) Prototype() schema.TypedPrototype {
+	return bindnode.Prototype(e, exampleType())
+}
+
+func (*Example) Tag() string {
+	return exampleTag
+}
+
+func exampleGoldenNode(t *testing.T) datamodel.Node {
```
Comment (fabiobozzo) [inline internal/envelope/example_test.go:87]: What's a "golden" node?

File: internal/envelope/ipld.go [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+
+		if kStr == "h" {
+			varsigHeaderNode = v
+
+			continue
+		}
+
+		if strings.HasPrefix(kStr, "ucan/") {
```
Comment (fabiobozzo) [inline internal/envelope/ipld.go:175]: `ucan/` tag prefix is part of the spec, isn't it?

**Review by fabiobozzo — APPROVED** [review]

File: internal/envelope/ipld.go (hunk 3 above)
Comment (smoyer64) [reply internal/envelope/ipld.go:175]: Just an IPLD node that's created from a golden file for testing.

**Pull request merged**